naga/src/front/glsl/parser/declarations.rs

```diff
@@ -251,7 +251,12 @@ impl<'source> ParsingContext<'source> {
                     init.and_then(|expr| ctx.ctx.lift_up_const_expression(expr).ok());
                 late_initializer = None;
             } else if let Some(init) = init {
-                if ctx.is_inside_loop || !ctx.ctx.local_expression_kind_tracker.is_const(init) {
+                if ctx.is_inside_loop
+                    || !ctx
+                        .ctx
+                        .local_expression_kind_tracker
+                        .is_const_or_override(init)
+                {
```

Comment:
```suggestion
                if ctx.is_inside_loop || !ctx.ctx.local_expression_kind_tracker.is_const(init) {
```

I think we should keep this as it was since GLSL doesn't have overrides.